<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1,viewport-fit=cover"><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=generator content="Hugo 0.149.0"><meta name=author content="Subho Sankar Banerjee"><meta name=description content="Efficient and Scalable Workflows for Genomic Analyses"><meta property="og:url" content="https://ssbanerje.github.io/publications/didc2016/"><meta property="og:site_name" content="Subho Sankar Banerjee"><meta property="og:title" content="Efficient and Scalable Workflows for Genomic Analyses"><meta property="og:description" content="Recent growth in the volume of DNA sequence data and associated computational costs of extracting meaningful information warrants the need for efficient computational systems at-scale. In this work, we propose the Illinois Genomics Execution Environment (IGen), a framework for efficient and scalable genome analyses. The design philosophy of IGen is based on algorithmic analysis and extensive measurements on compute- and data-intensive genomic analyses workflows (such as variant discovery and genotyping analysis) executed on high-performance and cloud computing infrastructures. IGen leverages the advantages of existing designs and proposes new software improvements to overcome the inefficiencies we observe in our measurements. Based on these composite improvements, we demonstrate that IGen is able to accelerate the alignment from 13.1 hours to 10.8 hours (1.2x) and the variant from 10.1 hours to 1.25 hours (8x) calling on a single node, and its modular design scales efficiently in a parallel computing environment."><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="publications"><meta property="article:published_time" content="2016-05-31T00:00:00+00:00"><meta property="article:modified_time" content="2016-05-31T00:00:00+00:00"><meta itemprop=name content="Efficient and Scalable Workflows for Genomic Analyses"><meta itemprop=description content="Recent growth in the volume of DNA sequence data and associated computational costs of extracting meaningful information warrants the need for efficient computational systems at-scale. In this work, we propose the Illinois Genomics Execution Environment (IGen), a framework for efficient and scalable genome analyses. The design philosophy of IGen is based on algorithmic analysis and extensive measurements on compute- and data-intensive genomic analyses workflows (such as variant discovery and genotyping analysis) executed on high-performance and cloud computing infrastructures. IGen leverages the advantages of existing designs and proposes new software improvements to overcome the inefficiencies we observe in our measurements. Based on these composite improvements, we demonstrate that IGen is able to accelerate the alignment from 13.1 hours to 10.8 hours (1.2x) and the variant from 10.1 hours to 1.25 hours (8x) calling on a single node, and its modular design scales efficiently in a parallel computing environment."><meta itemprop=datePublished content="2016-05-31T00:00:00+00:00"><meta itemprop=dateModified content="2016-05-31T00:00:00+00:00"><meta itemprop=wordCount content="147"><meta name=twitter:card content="summary"><meta name=twitter:title content="Efficient and Scalable Workflows for Genomic Analyses"><meta name=twitter:description content="Recent growth in the volume of DNA sequence data and associated computational costs of extracting meaningful information warrants the need for efficient computational systems at-scale. In this work, we propose the Illinois Genomics Execution Environment (IGen), a framework for efficient and scalable genome analyses. The design philosophy of IGen is based on algorithmic analysis and extensive measurements on compute- and data-intensive genomic analyses workflows (such as variant discovery and genotyping analysis) executed on high-performance and cloud computing infrastructures. IGen leverages the advantages of existing designs and proposes new software improvements to overcome the inefficiencies we observe in our measurements. Based on these composite improvements, we demonstrate that IGen is able to accelerate the alignment from 13.1 hours to 10.8 hours (1.2x) and the variant from 10.1 hours to 1.25 hours (8x) calling on a single node, and its modular design scales efficiently in a parallel computing environment."><meta name=theme-color content="#141f33"><title>Efficient and Scalable Workflows for Genomic Analyses | Subho Sankar Banerjee</title><link rel=canonical href=/publications/didc2016/><link rel=icon href=/img/favicon.ico><link rel="shortcut icon" href=/img/favicon.ico><link rel=apple-touch-icon href=/img/favicon.ico><link rel=stylesheet href=/css/style.min.a5fd5df6db3eafce9b1c5f5fabeadc25c3d64c8798012ac48a13c5fb32861ce7.css><link href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,400;0,700;1,400;1,700&amp;family=Inconsolata:ital,wght@0,400;0,700;1,400;1,700&amp;family=Eczar:ital,wght@0,400;0,700;1,400;1,700&display=swap" crossorigin=anonymous type=text/css rel=stylesheet><link href=https://unpkg.com/academicons@1.8.6/css/academicons.min.css crossorigin=anonymous type=text/css rel=stylesheet rel=preload as=style onload='this.onload=null,this.rel="stylesheet"'><link href=https://unpkg.com/components-font-awesome@5.9.0/css/all.min.css crossorigin=anonymous type=text/css rel=stylesheet rel=preload as=style onload='this.onload=null,this.rel="stylesheet"'></head><body><div class="columns is-gapless is-marginless"><div class="column is-one-quarter-tablet is-narrow-desktop" id=sidebar><div class=padded-sidebar><section class=section><figure class="image container" style=margin-bottom:2rem><img class=is-rounded src=/img/me_sq.png alt=Avatar></figure><div class="sidebar-content has-text-centered"><h1 class="title is-1"><a href=/>Subho Sankar Banerjee</a></h1><p class="has-text-white-ter has-text-centered">Software Engineer @ Google</p><ul class="icon-list is-clearfix"><li><a href=mailto:remove_this_if_human_ssbaner2@illinois.edu><i class="fas fa-envelope-open fa-fw"></i></a></li><li><a href="https://illinois.edu/map/view?buildingId=148"><i class="fas fa-map-marker-alt fa-fw"></i></a></li><li><a href="https://scholar.google.com/citations?user=jvV04p3aXgMC"><i class="ai ai-google-scholar fa-fw"></i></a></li><li><a href="https://dblp.uni-trier.de/pers/hd/b/Banerjee:Subho_S="><i class="ai ai-dblp fa-fw"></i></a></li><li><a href=https://www.linkedin.com/in/ssbanerjee><i class="fab fa-linkedin fa-fw"></i></a></li><li><a href=https://github.com/ssbanerje><i class="fab fa-github fa-fw"></i></a></li></ul></div></section></div></div><div class=column><div id=content class=has-text-justified><section class=section><nav class=breadcrumb aria-label=breadcrumbs><ul><li><a href=/><span class="icon is-small"><i class="fas fa-home" aria-hidden=true></i></span><span>Home</span></a></li><li><a href=/publications/>Publications</a></li><li class=is-active><a href=# aria-current=page>This Page</a></li></ul></nav></section><section class=section><div class=has-text-left><h1 class="title is-1 no-border">Efficient and Scalable Workflows for Genomic Analyses</h1><h3 class="title is-3">Subho S. Banerjee, Arjun P. Athreya, Liudmila S. Mainzer, C. Victor Jongeneel, and Wen-Mei Hwu, Zbigniew T. Kalbarczyk, and Ravishankar K. Iyer</h3><h3 class="title is-3 has-text-grey"><i>DIDC 2016 (Colocated with HPDC 2016)</i></h3></div><br><hr><ul class="single-paper-links is-size-3"><li><a href="http://dl.acm.org/citation.cfm?id=2912156" target=_blank rel=noreferrer rel=noopener><i class="ai ai-acmdl" aria-hidden=true></i>&nbsp;DOI</a></li><li><a href=/publications/didc2016/Paper.pdf target=_blank rel=noreferrer rel=noopener><i class="far fa-file-pdf" aria-hidden=true></i>&nbsp;Paper</a></li><li><a href=/publications/didc2016/Slides.pdf target=_blank rel=noreferrer rel=noopener><i class="far fa-file-pdf" aria-hidden=true></i>&nbsp;Slides</a></li></ul></section><section class=section><h2 class="title is-2">Abstract</h2><div class=content><p>Recent growth in the volume of DNA sequence data and associated computational costs of extracting
meaningful information warrants the need for efficient computational systems at-scale. In this work,
we propose the Illinois Genomics Execution Environment (IGen), a framework for efficient and
scalable genome analyses. The design philosophy of IGen is based on algorithmic analysis and
extensive measurements on compute- and data-intensive genomic analyses workflows (such as variant
discovery and genotyping analysis) executed on high-performance and cloud computing infrastructures.
IGen leverages the advantages of existing designs and proposes new software improvements to overcome
the inefficiencies we observe in our measurements. Based on these composite improvements, we
demonstrate that IGen is able to accelerate the alignment from 13.1 hours to 10.8 hours (1.2x) and
the variant from 10.1 hours to 1.25 hours (8x) calling on a single node, and its modular design
scales efficiently in a parallel computing environment.</p></div></section><section class=section style=margin-bottom:1rem><h2 class="title is-2">Citation
<a href=/publications/didc2016/ref.bib style=font-size:80%><i class="far fa-arrow-alt-circle-down"></i></a></h2><div class=highlight><pre tabindex=0 style=background-color:#f0f3f3;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bibtex data-lang=bibtex><span style=display:flex><span><span style=color:#0a8;font-weight:700>@inproceedings</span>{<span style=color:#99f>Banerjee2016</span>,
</span></span><span style=display:flex><span> <span style=color:#309>author</span> = <span style=color:#c30>{Banerjee, Subho S. and Athreya, Arjun P. and Mainzer, Liudmila S. and Jongeneel, C. Victor and Hwu, Wen-Mei and Kalbarczyk, Zbigniew T. and Iyer, Ravishankar K.}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>title</span> = <span style=color:#c30>{Efficient and Scalable Workflows for Genomic Analyses}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>booktitle</span> = <span style=color:#c30>{Proceedings of the ACM International Workshop on Data-Intensive Distributed Computing}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>series</span> = <span style=color:#c30>{DIDC &#39;16}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>year</span> = <span style=color:#c30>{2016}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>isbn</span> = <span style=color:#c30>{978-1-4503-4352-7}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>location</span> = <span style=color:#c30>{Kyoto, Japan}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>pages</span> = <span style=color:#c30>{27--36}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>numpages</span> = <span style=color:#c30>{10}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>url</span> = <span style=color:#c30>{http://doi.acm.org/10.1145/2912152.2912156}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>doi</span> = <span style=color:#c30>{10.1145/2912152.2912156}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>acmid</span> = <span style=color:#c30>{2912156}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>publisher</span> = <span style=color:#c30>{ACM}</span>,
</span></span><span style=display:flex><span> <span style=color:#309>address</span> = <span style=color:#c30>{New York, NY, USA}</span>,
</span></span><span style=display:flex><span>} 
</span></span></code></pre></div></section><section class=section style=margin-bottom:1rem><h2 class="title is-2">Related Projects</h2><ul class=bulleted-list><li><a href=/projects/learnedsystems/>Intelligence Augmented Compter Systems</a></li></ul></section></div><footer class=footer><div class="content has-text-left has-text-dark"><ul><li><span class="icon has-text-info"><i class="fas fa-heading fa-fw"></i></span> Powered by <a href=https://www.gohugo.io>Hugo</a></li><li><span class="icon has-text-link"><i class="fas fa-pencil-alt fa-fw"></i></span> Last updated 08/29/2025</li><li><a href=https://ssbanerje.github.io/index.xml><span class="icon has-text-orange"><i class="fas fa-rss fa-fw"></i></span> Feed</a></li></ul></div></footer></div><div class="column is-1 is-hidden-touch" id=right></div></div><script src=https://unpkg.com/quicklink@2.0.0/dist/quicklink.umd.js integrity=sha384-0nZ9FdSjDH2+tAS8yrmxpBHVMFs28rYR4bDVPFEoQqfzvezs/7OW7ebNI7CrWAg7 crossorigin=anonymous async></script><script>window.addEventListener("load",()=>{quicklink.listen({ignores:[e=>e.includes(".pdf"),e=>e.includes(".bib"),(e,t)=>t.hasAttribute("noprefetch")]})})</script></body></html>